language/move-prover/doc/user/spec-lang.md

```diff
@@ -92,11 +92,17 @@ discussed in the following sections.
 
 The type system of MSL is similar to that of Move, with the following differences:
 
-- All integer types of Move (`u8`, `u64`, and `u128`) are treated as the same type. In
+- There are two types of encodings for integer types: `num` and `bv` (bit vector).
+  If an integer (either a constant or a variable) does not involve in any bitwise operations directly or indirectly,
+  regardless of its type in Move (`u8`, `u16`, `u32`, `u64`, `u128` and `u256`), it is treated as the same type. In
   specifications, this type is called `num`, which is an arbitrary precision *signed* integer type.
   When MSL refers to a Move name which represents an `u8` or such, it will be automatically widened
   to `num`. This allows writing MSL expressions like `x + 1 <= MAX_U128` or `x - y >= 0` without
   needing to worry about overflow or underflow.
+  Different from `num`, `bv` cannot and does not need to be explicitly used in specifications: if an integer involves in bitwise operations such as `&`, `|` or `^`, it will be automatically encoded as `bv`at the backend.
```

Comment:
The automatic detection for encoding is already done or something to be done?

Reply:
It is already done